tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/GenIncOptimizer.scala

```diff
@@ -696,7 +705,7 @@ abstract class GenIncOptimizer(semantics: Semantics,
     private[this] var _deleted: Boolean = false
 
     var lastInVersion: Option[String] = None
-    var lastOutVersion: Int = 0
+    var lastOutVersion: Int = randomByThread.get.nextInt()
```

Comment:
Basically, here we need something like:

```scala
var lastOutVersion: Int = outVersions.getOrElse(owner.encodedName, 0).getOrElse(encodedName, 0)
```

Further, something that stores the outVersion when we bump it.